blockchain/storage/leveldb/leveldb.go

```diff
@@ -4,10 +4,51 @@ import (
 	"fmt"
 
 	"github.com/dogechain-lab/dogechain/blockchain/storage"
+	"github.com/dogechain-lab/dogechain/helper/common"
 	"github.com/hashicorp/go-hclog"
 	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/filter"
+	leveldbopt "github.com/syndtr/goleveldb/leveldb/opt"
 )
 
+const (
+	// minCache is the minimum memory allocate to leveldb
+	// half write, half read
+	minCache = 16
+
+	// minHandles is the minimum number of files handles to leveldb open files
+	minHandles = 16
+
+	DefaultCache   = 16  // 16 MiB
+	DefaultHandles = 512 // files handles to leveldb open files
+
+	DefaultBloomKeyBits = 16 // bloom filter bits
+
+	DefaultCompactionTableSize = 2  // 2 MiB
+	DefaultCompactionTotalSize = 10 // 10 MiB
+	DefaultNoSync              = false
+)
+
+type Options struct {
+	CacheSize           int
+	Handles             int
+	BloomKeyBits        int
+	CompactionTableSize int
+	CompactionTotalSize int
+	NoSync              bool
+}
+
+func NewDefaultOptons() *Options {
+	return &Options{
+		CacheSize:           minCache,
+		Handles:             minHandles,
+		BloomKeyBits:        DefaultBloomKeyBits,
+		CompactionTableSize: DefaultCompactionTableSize,
+		CompactionTotalSize: DefaultCompactionTotalSize,
+		NoSync:              DefaultNoSync,
+	}
+}
+
 // Factory creates a leveldb storage
 func Factory(config map[string]interface{}, logger hclog.Logger) (storage.Storage, error) {
 	path, ok := config["path"]
```

Comment:
Can we refactor this repeat format?

Reply:
Builder design pattern ?

Reply:
That would be a better choice.

Reply:
reimplement the interface, use builder pattern

Reply:
Great job. 🙌